### ISSUE TITLE
Bug - Error when click on edit HTML button

### ISSUE THREAD

**Issue by gadlol** [bootstrap-wysiwyg/bootstrap3-wysiwyg#105]
Although you can display & edit HTML fine I get error in the javascript console.

In Goggle chrome I get : Uncaught TypeError: Cannot read property 'ownerDocument' of undefined 
In Firefox: TypeError: this.element is undefined

This occurs when you click on the edit HTML button.

Using the latest version of bootstrap3-wysiwyg

Chrome Console Screenshot

![screenshot from 2014-10-13 15 28 15](https://cloud.githubusercontent.com/assets/518451/4613336/894c1fa8-52d4-11e4-9e06-e67b4978fb00.png)

Firefox Console Screenshot

![screenshot from 2014-10-13 15 29 55](https://cloud.githubusercontent.com/assets/518451/4613354/bb442cd0-52d4-11e4-9a5e-b40bbae04a53.png)

**Comment by apocratus** [comment]
I have the same problem, even worse, on Safari (Mac) it breaks the "createLink" functionality.

Error: "typeError: undefined is not an object (evaluating 'this.element.ownerdocument')"

Seems to be the same bug as this one:
https://github.com/jhollingworth/bootstrap-wysihtml5/issues/48

Although I cannot get the proposed fix to work with this fork.

I could really use a fix for this as the client is working on Safari on a Mac and is hesitant to use another browser.

**Comment by abishekrsrikaanth** [comment]
Is there a fix for this issue?

**Comment by apocratus** [comment]
I haven't found a proper fix yet, but it might help to know that I'm now using an older version which doesn't seem to have the problem: 
https://github.com/bootstrap-wysiwyg/bootstrap3-wysiwyg/blob/f008b06b128a79158e5722570de18330e477df73/dist/bootstrap3-wysihtml5.all.min.js

(bootstrap3-wysihtml5-bower 2014-06-11)

**Comment by pavennt** [comment]
Yes i same eror "Uncaught TypeError: Cannot read property 'ownerDocument' of undefined" . This on version (bootstrap3-wysihtml5-bower 2015-05-08).

**Comment by volmer** [comment]
Same here. Isn't there any fix yet?

![image](https://cloud.githubusercontent.com/assets/301187/9559962/b565fc7c-4dcd-11e5-842e-717fa46912b7.png)

**Comment by JIAZHEN** [comment]
I got the same error, could anyone suggest a solution? @Waxolunist

**Comment by rubyrider** [comment]
I get it working by changing that 12001 line to:
this.element.ownerDocument.body === this.element
https://github.com/bootstrap-wysiwyg/bootstrap3-wysiwyg/blob/master/dist/bootstrap3-wysihtml5.all.js#L12001

I found this.element.ownerDocument.querySelector(":focus") returning null value.
Its working for the link part.

**Comment by volmer** [comment]
@rubyrider can you submit a PR?

**Comment by JarJak** [comment]
Same here.

**Comment by volmer** [comment]
I submitted #162. It won't fix the issue but might let people avoid this fork since it's no longer maintained.

**Comment by tsgold8899** [comment]
Is there any solution for the problem?

**Comment by JarJak** [comment]
@tsgold8899 Get older version or switch to https://github.com/summernote/summernote

**Comment by berhalak** [comment]
Well, try changing line number 14705 in all.js

`          dialog.on('hide', function() {
            $(this.container).modal('hide');
            setTimeout(editor.composer.focus, 0);
          });`

to
`  dialog.on('hide', function() {
            $(this.container).modal('hide');
            setTimeout(function () {
                editor.composer.focus.call(editor.composer);
            }, 0);
          });`
I think, here the scope is being lost.

**Comment by firmanhcf** [comment]
@rubyrider its work

**Comment by flymorn** [comment]
@berhalak Thanks! it works.

**Comment by Palanikumarsprw** [comment]
Not working for me

**Comment by ozzpy** [comment]
wysihtml5 v0.3.0
on file bootstrap-wysihtml5.js, comment line 279 //self.editor.currentView.element.focus(false);
